ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py

```diff
@@ -323,15 +325,28 @@ def _wan_load_balance_policies(self) -> list:
         control_plane_virtual_topology = get(self._hostvars, "wan_virtual_topologies.control_plane_virtual_topology", default={})
         name = get(control_plane_virtual_topology, "name", default="CONTROL-PLANE-PROFILE")
         wan_load_balance_policies = [self._generate_wan_load_balance_policy(f"{name}_LB", control_plane_virtual_topology, default_all=True)]
-        for avt_policy in get(self._hostvars, "wan_virtual_topologies.policies", []):
+        for avt_policy in self._filtered_wan_policies:
+            policy_name = avt_policy["name"] if avt_policy["name"] != self._default_vrf_policy["name"] else avt_policy["realname"]
             for application_virtual_topology in get(avt_policy, "application_virtual_topologies", []):
                 # TODO add internet exit once supported
-                name = get(application_virtual_topology, "name", default=f"{avt_policy['name']}_{application_virtual_topology['application_profile']}")
-                wan_load_balance_policies.append(self._generate_wan_load_balance_policy(f"{name}_LB", application_virtual_topology))
+                name = get(application_virtual_topology, "name", default=f"{policy_name}_{application_virtual_topology['application_profile']}")
+                append_if_not_duplicate(
+                    list_of_dicts=wan_load_balance_policies,
+                    primary_key="name",
+                    new_dict=self._generate_wan_load_balance_policy(f"{name}_LB", application_virtual_topology),
```

Comment:
I think ppl normally prefix the names with a short form of the type. We also need to be consistent with _ vs - and upper/lower case.

Reply:
problem is that we don't control the `name` part so they can do whatever they want..

Reply:
will move it to prefix